polymorphic/tests.py

```diff
@@ -1325,6 +1341,44 @@ def func():
         # Ensure no queries are made using the default database.
         self.assertNumQueries(0, func)
 
+    def test_unknown_issue(self):
+        user_a = UserProfile.objects.create(name='a')
+        user_b = UserProfile.objects.create(name='b')
+        user_c = UserProfile.objects.create(name='c')
+
+        team1 = Team.objects.create(team_name='team1')
+        team1.user_profiles.add(user_a, user_b, user_c)
+        team1.save()
+
+        team2 = Team.objects.create(team_name='team2')
+        team2.user_profiles.add(user_c)
+        team2.save()
+
+        # without prefetch_related, the test passes
+        my_teams = Team.objects.filter(user_profiles=user_c).prefetch_related('user_profiles').distinct()
+
+        print(my_teams[0].user_profiles.all())
+        print(my_teams[1].user_profiles.all())
+        self.assertEqual(len(my_teams[0].user_profiles.all()), 3)
```

Comment:
This doesn't look right; you're not ordering you QuerySet, so the first result returned by the DB might be `team2`. It'll probably fail, occasionally.